watcherx/directory.go

```diff
@@ -0,0 +1,120 @@
+package watcherx
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+)
+
+func listSubDirsDepth(parent string, depth int) (dirs []string, errs []error) {
+	if depth == 0 {
+		return
+	}
+	entries, err := ioutil.ReadDir(parent)
+	if err != nil {
+		return dirs, append(errs, errors.WithStack(err))
```

Comment:
Why not just return on an error immediately, given that you return directly here as well: https://github.com/ory/x/pull/180/files#diff-61f10616b3d8e8d2b3d31a972c41c59eR41